extensions/positron-r/src/provider.ts

```diff
@@ -25,123 +25,102 @@ export const R_DOCUMENT_SELECTORS = [
-	if (pathBinary && !binaries.has(pathBinary)) {
-		binaries.set(pathBinary, BinarySource.PATH);
-	}
+	const userMoreBinaries = discoverAdHocBinaries(userRBinaries());
+	updateBinaries(userMoreBinaries);
 
-	// make sure we include the "current" version of R, for some definition of "current"
-	// we've probably already discovered it, but we still want to single it out, so that we mark
-	// that particular R installation as the current one
-	const curBin = await findCurrentRBinary();
-	if (curBin) {
-		rInstallations.push(new RInstallation(curBin, true));
-		binaries.delete(curBin);
+	// (Try to) promote each RBinary to a proper RInstallation
+	let rInstallations: Array<RInstallation> = [];
+	if (currentBinaries.length > 0) {
+		const currentBinary = currentBinaries[0];
+		rInstallations.push(new RInstallation(currentBinary.path, true, currentBinary.reasons));
+		binaries.delete(currentBinary.path);
 	}
 
-	binaries.forEach((source, bin) => {
-		rInstallations.push(new RInstallation(bin));
+	binaries.forEach(rbin => {
+		rInstallations.push(new RInstallation(rbin.path, false, rbin.reasons));
 	});
 
-	// TODO: possible location to tell the user why certain R installations are being omitted from
-	// the interpreter drop-down and, in some cases, offer to help fix the situation:
-	// * version < minimum R version supported by positron-r
-	// * (macOS only) version is not orthogonal and is not the current version of R
-	// * invalid R installation
+	const rejectedRInstallations: Array<RInstallation> = [];
 	rInstallations = rInstallations
 		.filter(r => {
-			if (!r.valid) {
-				LOGGER.info(`Filtering out ${r.binpath}: invalid R installation.`);
-				return false;
-			}
-			return true;
-		})
-		.filter(r => {
-			if (!(r.current || r.orthogonal)) {
-				LOGGER.info(`Filtering out ${r.binpath}: not current and also not orthogonal.`);
-				return false;
-			}
-			return true;
-		})
-		.filter(r => {
-			if (!r.supported) {
-				LOGGER.info(`Filtering out ${r.binpath}: version is < ${MINIMUM_R_VERSION}`);
+			if (!r.usable) {
+				LOGGER.info(`Filtering out ${r.binpath}, reason: ${friendlyReason(r.reasonRejected)}.`);
+				rejectedRInstallations.push(r);
 				return false;
 			}
 			return true;
 		});
 
-	// FIXME? should I explicitly check that there is <= 1 R installation
-	// marked as 'current'?
+	if (rejectedRInstallations.length > 0) {
+		if (rInstallations.length === 0) {
+			LOGGER.warn(`All discovered R installations are unusable by Positron.`);
+			LOGGER.warn('Learn more about R discovery at https://positron.posit.co/r-installations.html');
+			const showLog = await positron.window.showSimpleModalDialogPrompt(
+				vscode.l10n.t('All discovered R installations are unusable by Positron.'),
+				vscode.l10n.t('Learn more about R discovery at <br><a href="https://positron.posit.co/r-installations.html">https://positron.posit.co/r-installations.html</a>'),
+				vscode.l10n.t('View logs'),
+				vscode.l10n.t('Dismiss')
+			);
```

Comment:
How this currently looks:

<img width="448" alt="Screenshot 2024-12-09 at 2 33 18 PM" src="https://github.com/user-attachments/assets/551c6f2b-5d09-4742-8138-02ff7b88b293">

I will open a separate issue about improving `positron.window.showSimpleModelDialogPrompt()`, as per a related slack discussion. In the meantime, I'm making do.